Tests/LoadFeedFromRemoteUseCaseTests.swift

```diff
@@ -24,108 +24,110 @@ class LoadFeedFromRemoteUseCaseTests: XCTestCase {
-//		})
-//	}
-//
-//	func test_load_doesNotDeliverResultAfterSUTInstanceHasBeenDeallocated() {
-//		let url = URL(string: "http://any-url.com")!
-//		let client = HTTPClientSpy()
-//		var sut: RemoteFeedLoader? = RemoteFeedLoader(url: url, client: client)
-//
-//		var capturedResults = [RemoteFeedLoader.Result]()
-//		sut?.load { capturedResults.append($0) }
+	func test_loadTwice_requestsDataFromURLTwice() {
+		let url = URL(string: "https://a-given-url.com")!
+		let (sut, client) = makeSUT(url: url)
+
+		sut.load { _ in }
+		sut.load { _ in }
+
+		XCTAssertEqual(client.requestedURLs, [url, url])
+	}
+
+	func test_load_deliversConnectivityErrorOnClientError() {
+		let (sut, client) = makeSUT()
+
+		expect(sut, toCompleteWith: .failure(.connectivity), when: {
+			let clientError = NSError(domain: "Test", code: 0)
+			client.complete(with: clientError)
+		})
+	}
+
+	func test_load_deliversInvalidDataErrorOnNon200HTTPResponse() {
+		let (sut, client) = makeSUT()
+
+		let samples = [199, 201, 300, 400, 500]
+
+		samples.enumerated().forEach { index, code in
+			expect(sut, toCompleteWith: .failure(.invalidData), when: {
+				let json = makeItemsJSON([])
+				client.complete(withStatusCode: code, data: json, at: index)
+			})
+		}
+	}
+
+	func test_load_deliversInvalidDataErrorOn200HTTPResponseWithInvalidJSON() {
+		let (sut, client) = makeSUT()
+
+		expect(sut, toCompleteWith: .failure(.invalidData), when: {
+			let invalidJSON = Data("invalid json".utf8)
+			client.complete(withStatusCode: 200, data: invalidJSON)
+		})
+	}
+
 //
-//		sut = nil
-//		client.complete(withStatusCode: 200, data: makeItemsJSON([]))
+	func test_load_deliversInvalidDataErrorOn200HTTPResponseWithPartiallyValidJSONItems() {
+		let (sut, client) = makeSUT()
+
+		let validItem = makeItem(
+			id: UUID(),
+			imageURL: URL(string: "http://another-url.com")!
+		).json
+
+		let invalidItem = ["invalid": "item"]
+
+		let items = [validItem, invalidItem]
+
+		expect(sut, toCompleteWith: .failure(.invalidData), when: {
+			let json = makeItemsJSON(items)
+			client.complete(withStatusCode: 200, data: json)
+		})
+	}
+
+	func test_load_deliversSuccessWithNoItemsOn200HTTPResponseWithEmptyJSONList() {
+		let (sut, client) = makeSUT()
+
+		expect(sut, toCompleteWith: .success([]), when: {
+			let emptyListJSON = makeItemsJSON([])
+			client.complete(withStatusCode: 200, data: emptyListJSON)
+		})
+	}
+
+	func test_load_deliversSuccessWithItemsOn200HTTPResponseWithJSONItems() {
+		let (sut, client) = makeSUT()
+
+		let item1 = makeItem(
+			id: UUID(),
+			imageURL: URL(string: "http://a-url.com")!)
+
+		let item2 = makeItem(
+			id: UUID(),
+			description: "a description",
+			location: "a location",
+			imageURL: URL(string: "http://another-url.com")!)
+
+		let items = [item1.model, item2.model]
+
+		expect(sut, toCompleteWith: .success(items), when: {
+			let json = makeItemsJSON([item1.json, item2.json])
+			client.complete(withStatusCode: 200, data: json)
+		})
+	}
+
 //
```

Comment:
Please remove extraneous commented lines on the `LoadFeedFromRemoteUseCaseTests` file.
```suggestion
		})
	}


```